src/main/java/com/zendesk/maxwell/MaxwellTableCache.java

```diff
@@ -22,7 +23,7 @@ public void processEvent(Schema schema, MaxwellFilter filter, TableMapEvent even
 
 			Table tbl = db.findTable(tblName);
 
-			if ( filter != null && filter.isTableBlacklisted(tblName) )
+			if ( filter != null && filter.isTableBlacklisted(tbl.getDatabase().getName(), tblName) )
```

Comment:
@jpeyper my eyes sort of glossed over this line, but in local testing I found that it's subject to crashing -- if the database is blacklisted, it's quite possible that `db.findTable()` will return `null` (if we didn't process the `CREATE TABLE` ddl), or the call to `schema.findDatabase` may even return null (if we didn't process the `CREATE DATABASE` ddl).  

Can you do an integration test that sets up a database blacklist and then runs through a few scenarios?  notably:

  - create DB
  - capture schema
  - blacklist DB
  - create table DB.tbl
  - insert into DB.tbl

and also

  - capture schema
  - blacklist DB
  - create DB
  - create tbl
  - insert into DB.tbl